packages/cli/src/WebhookServer.ts

```diff
@@ -1,7 +1,31 @@
 import { AbstractServer } from '@/AbstractServer';
 
 export class WebhookServer extends AbstractServer {
-	constructor() {
+	constructor(opts: {
+		shutdown: {
+			signal: AbortSignal;
+			timeoutInS: number;
+		};
+	}) {
 		super('webhook');
+
+		const { signal: shutdownSignal, timeoutInS } = opts.shutdown;
+
+		shutdownSignal.addEventListener(
+			'abort',
+			async () => {
+				await this.stop(timeoutInS);
+			},
+			{ once: true },
+		);
+	}
```

Comment:
Can't we move this code to AbstractServer as well?

Reply:
Moved it to AbstractServer